### PR TITLE
Add link to try out current Spyder on mybinder.org

### PR DESCRIPTION
**Pull request by yuvipanda**
## Description of Changes

Accessing [this mybinder.org link](https://mybinder.org/v2/gh/yuvipanda/spyder/binder-link?urlpath=/desktop)
will give you a Linux desktop environment with a desktop shortcut to Spyder.
Launching that will run `./bootstrap.py` and launch the master (or whatever
branch you are on) version of Spyder.

This lets people try out new branches and unreleased versions easily without
having to install anything locally.

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:

YuviPanda

### CODE REVIEW

**Comment by yuvipanda** [comment]
Note that the link I added in the README won't work until this PR is merged. In the meantime, you can test it out with https://mybinder.org/v2/gh/yuvipanda/spyder/binder-link?urlpath=/desktop

File: binder/environment.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+- sphinx
+- spyder-kernels >=1.8.1,<2.0.0
+- watchdog
+
+- spyder-kernels=1.*
+
+# Required for jupyter-desktop-server
+- websockify
+- pip:
+    - jupyter-desktop-server
```
Comment (goanpeca) [inline binder/environment.yml]: @yuvipanda could we also add the content ot `requirements/tests.txt` ad a message there as well?

Since we also want people to have those packages for testing installed.

Thanks!

File: binder/Desktop/Spyder.desktop [hunk 2]
```diff
@@ -0,0 +1,7 @@
+[Desktop Entry]
+Version=1.0
+Type=Application
+Name=Spyder (Development Version)
+Exec=/bin/bash -c "exec ${HOME}/bootstrap.py"
+# FIXME: Don't hardcode Path?
+Path=/home/jovyan/Desktop
```
Comment (goanpeca) [inline binder/Desktop/Spyder.desktop]: Could we add an icon? It would make it easier for easier for users to know what to click

Comment (goanpeca) [reply binder/Desktop/Spyder.desktop]: We could use this one.

https://github.com/spyder-ide/spyder/blob/master/img_src/spyder.ico

**Comment by yuvipanda** [comment]
Just tested out locally, I see the icon! \o/ 

image

I also see that matplotlib (From tests) is installed:

image

File: binder/environment.yml (hunk 1 above)
Comment (yuvipanda) [reply binder/environment.yml]: Done

File: binder/Desktop/Spyder.desktop (hunk 2 above)
Comment (yuvipanda) [reply binder/Desktop/Spyder.desktop]: Done

**Comment by goanpeca** [comment]
Wohooo! :-) , so is this good to merge then @yuvipanda ?

File: binder/apt.txt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+dbus-x11
+firefox
```
Comment (ccordoba12) [inline binder/apt.txt]: Do we need really to install Firefox? I mean, how much overhead does it add to the deployment time @yuvipanda?

File: binder/environment.yml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+- qdarkstyle >=2.7
+- qtawesome >=0.5.7
+- qtconsole >=4.6.0
+- qtpy >=1.5.0
+- rtree
+- sphinx
+- spyder-kernels >=1.8.1,<2.0.0
+- watchdog
+
+- spyder-kernels=1.*
```
Comment (ccordoba12) [inline binder/environment.yml]: I think this line is not necessary because it's covered by the one above (`spyder-kernels >=1.8.1,<2.0.0`).

File: binder/environment.yml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+- sympy
+- pillow
+- matplotlib
+- cython
+- flaky
+
+# Required for jupyter-desktop-server
+- websockify
+- pip:
+    - jupyter-desktop-server
```
Comment (goanpeca) [inline binder/environment.yml]: Why is this needed?

File: binder/Desktop/Spyder.desktop [hunk 6]
```diff
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Version=1.0
+Type=Application
+Name=Spyder (Development Version)
+Exec=/bin/bash -c "exec ${HOME}/bootstrap.py"
+# FIXME: Don't hardcode Path?
+Icon=/home/jovyan/img_src/spyder.svg
```
Comment (goanpeca) [inline binder/Desktop/Spyder.desktop]: this works ? 

~/img_src/spyder.svg

File: binder/apt.txt (hunk 3 above)
Comment (goanpeca) [reply binder/apt.txt]: This is cached, as long as dependencies do not change I think this does not add much overhead.

File: binder/environment.yml (hunk 5 above)
Comment (yuvipanda) [reply binder/environment.yml]: This is what provides the Linux desktop experience on the server

File: binder/Desktop/Spyder.desktop (hunk 6 above)
Comment (yuvipanda) [reply binder/Desktop/Spyder.desktop]: Expanding ~ or $HOME in .desktop files seems complicated, and a few things I tried didn't work. Is it ok to defer that to another PR?

File: binder/environment.yml (hunk 4 above)
Comment (yuvipanda) [reply binder/environment.yml]: Done

File: binder/apt.txt (hunk 3 above)
Comment (yuvipanda) [reply binder/apt.txt]: Good catch. Was leftover from another demo. Removed

File: binder/Desktop/Spyder.desktop (hunk 6 above)
Comment (goanpeca) [reply binder/Desktop/Spyder.desktop]: Yep its fine

**Comment by yuvipanda** [comment]
Thanks for the merge, @ccordoba12 / @goanpeca!

Shout out to @manics and @ryanlovett who did a lot of the initial work to enable this kinda stuff.

**Comment by ccordoba12** [comment]
Thanks @manics and @ryanlovett, we really appreciate it!
